pkg/ccip/provider/provider.go

```diff
@@ -20,26 +24,40 @@ var _ commontypes.CCIPProvider = &Provider{}
 const CCIPProviderName = "TONCCIPProvider"
 
 type Provider struct {
-	lggr logger.Logger
-	ca   ccipocr3.ChainAccessor
-	ct   ocr3types.ContractTransmitter[[]byte]
+	lggr  logger.Logger
+	ca    ccipocr3.ChainAccessor
+	ct    ocr3types.ContractTransmitter[[]byte]
+	codec ccipocr3.Codec
 
 	wg sync.WaitGroup
 	services.StateMachine
 }
 
-func NewCCIPProvider(lggr logger.Logger, txm txm.TxManager) (*Provider, error) {
+func NewCCIPProvider(lggr logger.Logger, chainSelector ccipocr3.ChainSelector, client ton.APIClientWrapped, txm txm.TxManager, logPoller logpoller.LogPoller) (*Provider, error) {
 	ct, err := ocr.NewCCIPTransmitter(txm, lggr)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create a CCIP ContractTransmitter %w", err)
+		return nil, fmt.Errorf("failed to create a CCIP ContractTransmitter: %w", err)
 	}
 
-	cp := &Provider{
-		lggr: logger.Named(lggr, CCIPProviderName),
-		ct:   ct,
+	codec := ccipocr3.Codec{
+		ChainSpecificAddressCodec: &codec.AddressCodec{},
+		CommitPluginCodec:         &codec.CommitPluginCodecV1{},
+		ExecutePluginCodec:        &codec.ExecutePluginCodecV1{},
+		TokenDataEncoder:          nil, // TODO(NONEVM-1460): implement
```

Comment:
@huangzhen1997 can you take a look, I couldn't find the token data encoder

Reply:
This is for CCTP, we can create use a noop version for now